### PR TITLE
#271 Support link handles to enable manipulation of links and detache…

### PR DESCRIPTION
**Pull request by tomlyn**
…d links

Fixes: #271 

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Review by caritaou — COMMENTED** [review]
I hit an error when trying this scenario: 
I dragged a node onto a detached link and it was able to associate it. I then dragged another node onto the canvas first, then dropped it onto the detached link and it connected. Clicking undo resulted in the following error:

image

link drop error

**Comment by caritaou** [comment]
Its hard to see the visual feedback when dropping a node from palette onto a selected link. Should dragging from the palette deselect links?
drop node on selected link

**Comment by tomlyn** [comment]
@caritaou Good catch on both of these. I have fixed them both. I also noticed that region select was not selecting links when `enableLinkSelection` is set to something other than `None` so I also fixed that one also.

File: canvas_modules/harness/assets/images/custom-canvases/detached-links/decorations/dragStateArrow.svg [hunk 1]
```diff
@@ -0,0 +1,2840 @@
...
+	<g i:extraneous="self">
+		<g>
+			<circle fill="#FFFFFF" cx="10" cy="10" r="10"/>
+			<path fill="#0F62FE" d="M10,2c4.41,0,8,3.59,8,8s-3.59,8-8,8s-8-3.59-8-8S5.59,2,10,2 M10,0C4.48,0,0,4.48,0,10s4.48,10,10,10
+				s10-4.48,10-10S15.52,0,10,0L10,0z"/>
+		</g>
+		<polygon fill="#0F62FE" points="8.59,16 7,14.51 11.82,10 7,5.49 8.59,4 15,10 		"/>
+	</g>
+</switch>
+<i:aipgf  id="adobe_illustrator_pgf" i:pgfEncoding="zstd/base64" i:pgfVersion="24">
```
Comment (caritaou) [inline canvas_modules/harness/assets/images/custom-canvases/detached-links/decorations/dragStateArrow.svg:30]: Are all these required for the svg image? or can it be cleaned up?

Comment (tomlyn) [reply canvas_modules/harness/assets/images/custom-canvases/detached-links/decorations/dragStateArrow.svg:30]: In Elyra Canvas they are just used in the sample applications so they are not shipped with the product. They do however exist in Modeler Flows product. I'd noticed this before so we already have this issue for that problem: 
https://github.ibm.com/NGP-TWC/wml-canvas-planning/issues/8451 

I also opened one in wdp-abstract-canvas as a reminder to add those images into the sample apps when we get them back from design.
https://github.ibm.com/NGP-TWC/wdp-abstract-canvas/issues/2955

**Review by caritaou — APPROVED** [review]

**Pull request merged**